src/components/Signup.tsx

```diff
@@ -0,0 +1,252 @@
+    onRedirectToLogin?: VoidFunction
+    presetEmail?: string
+    appearance?: SignupAppearance
+}
+
+export type SignupAppearance = {
+    options?: {
+        headerContent?: ReactNode
+        displayLogo?: boolean
+        divider?: ReactNode | boolean
+        firstNameLabel?: ReactNode
+        lastNameLabel?: ReactNode
+        emailLabel?: ReactNode
+        usernameLabel?: ReactNode
+        passwordLabel?: ReactNode
+        submitButtonContent?: ReactNode
+        loginButtonContent?: ReactNode
+    }
+    elements?: {
+        Container?: ElementAppearance<ContainerProps>
+        Logo?: ElementAppearance<ImageProps>
+        Header?: ElementAppearance<H3Props>
+        Divider?: ElementAppearance<DividerProps>
+        FirstNameInput?: ElementAppearance<InputProps>
+        LastNameInput?: ElementAppearance<InputProps>
+        UsernameInput?: ElementAppearance<InputProps>
+        EmailInput?: ElementAppearance<InputProps>
+        PasswordInput?: ElementAppearance<InputProps>
+        SocialButton?: ElementAppearance<ButtonProps>
+        SubmitButton?: ElementAppearance<ButtonProps>
+        LoginButton?: ElementAppearance<ButtonProps>
+        ErrorMessage?: ElementAppearance<AlertProps>
+    }
+}
+
+export const Signup = ({ onSuccess, onRedirectToLogin, presetEmail, appearance }: SignupProps) => {
+    const { config } = useConfig()
+
+    return (
+        <div data-contain="component">
+            <Container appearance={appearance?.elements?.Container}>
+                {appearance?.options?.displayLogo !== false && config && (
+                    <div data-contain="logo">
+                        <Image
+                            src={config.logo_url}
+                            alt={config.site_display_name}
+                            appearance={appearance?.elements?.Logo}
+                        />
+                    </div>
+                )}
+                <div data-contain="header">
+                    <H3 appearance={appearance?.elements?.Header}>{appearance?.options?.headerContent || "Signup"}</H3>
+                </div>
+                <SignInOptions config={config} buttonAppearance={appearance?.elements?.SocialButton} />
+                {config &&
+                    config.has_password_login &&
+                    config.has_any_social_login &&
+                    appearance?.options?.divider !== false && (
+                        <SignInDivider
+                            appearance={appearance?.elements?.Divider}
+                            options={appearance?.options?.divider}
+                        />
+                    )}
+                {config && config.has_password_login && (
+                    <SignupForm config={config} onSuccess={onSuccess} presetEmail={presetEmail} />
+                )}
+                <BottomLinks onRedirectToLogin={onRedirectToLogin} appearance={appearance} />
+            </Container>
+        </div>
+    )
+}
+
+export type SignupOptions = {
```

Comment:
Doesn't this come from the API?

Reply:
[fix](https://github.com/PropelAuth/react/pull/11/commits#:~:text=697a7e1-,fix%20type%20import,-joshdschneider%20committed%20now)